src/resources/pandoc/datadir/init.lua

```diff
@@ -2089,7 +2093,8 @@ quarto = {
   json = json,
   base64 = base64,
   log = logging,
-  version = version()
+  version = version(),
+  cli_path = cli_path()
```

Comment:
How about making a `config` object here, and then adding bin_path to the object? That would match the TS API and would allow us to expose other related things in the future.

Reply:
In addition, you would then create a `src/resources/lua-types/quarto/config.lua` file with the type information for the VS Code extension.

Reply:
Oh I see, creating a new namespace. 🤔 

`quarto.version` was made with `quartoConfig.version()`, and not in a config part. So I followed that. All the more because  also `quartoConfig.binPath()` already exists so adding a `quarto.config.bin_path()` in Lua would need to match another `quartoConfig` object ... or start `quartoConfig.config.bin_path()` ? Not sure. 

Though on Lua side, having `quarto.config` to store this seems a better idea that at first level. I can move `quarto.cli_path()` to `quarto.config.cli_path`, and maybe have `quarto.config.version` too keeping the  other for backward compatibility. We could also make sure that all `quarto.config.` objects are functions without breaking old code. 

> In addition, you would then create a src/resources/lua-types/quarto/config.lua file with the type information for the VS Code extension.

Thanks, I was not aware of this. We'll do.

Reply:
I have done some changes to take into account

Reply:
> So I followed that. All the more because also `quartoConfig.binPath()` already exists so adding a `quarto.config.bin_path()` in Lua would need to match another `quartoConfig` object ... or start `quartoConfig.config.bin_path()`

In Lua we have all of our public APIs under `quarto`, so if we use `quarto.config` to be the Lua equivalent of `quartoConfig`, then we both get a public API, and a slightly better named version than `quartoConfig.config...`

> By "That would match the TS API", you are referring to `quartoConfig` right ?

Yes.

> Though on Lua side, having `quarto.config` to store this seems a better idea that at first level. I can move `quarto.cli_path()` to `quarto.config.cli_path`, and maybe have `quarto.config.version` too keeping the other for backward compatibility. We could also make sure that all `quarto.config.` objects are functions without breaking old code.

Yes, they should be functions, again because we want uniformity with the TS API.